gno.land/pkg/integration/testing_integration.go

```diff
@@ -273,3 +338,34 @@ func tsValidateError(ts *testscript.TestScript, cmd string, neg bool, err error)
 		}
 	}
 }
+
+type envSetter interface {
+	Setenv(key, value string)
+}
+
+// createAccount creates a new account with the given name and adds it to the keybase.
+func createAccount(env envSetter, kb keys.Keybase, accountName string) (balance gnoland.Balance, err error) {
```

Comment:
You don't need the keybase for this functionality, it's super slow.

Check out what I've done in #1212:

https://github.com/gnolang/gno/blob/a5dedd6020edb6171b8cf10e0e733f87406430e3/tm2/pkg/crypto/keys/client/add.go#L353-L379

Reply:
Help me understand why the call to `Keybase.CreateAccount` isn't needed.  I looked at the code you linked to, and before it calls into that generate addresses, it calls `Keybase.CreateAccount`.  As far as I can tell, this writes the new account info to the keybase store which is then read from each time `maketx` is used.  How else would the data get into the keybase store with out using `CreateAccount`?

Or are you suggesting that _instead of_ creating new keys for each account, it uses the same mnemonic to derive multiple account addresses and use those? In that case how could I map account names to the derived addresses? Like if I derived an address and mapped it to the name `test2`, what is the argument that gets passed in to `gnokey`?

Sorry if all of that is wrong; I'm just not sure what you're getting at here, maybe due to my relative unfamiliarity with this area of the code.

Reply:
> Help me understand why the call to Keybase.CreateAccount isn't needed. I looked at the code you linked to, and before it calls into that generate addresses, it calls Keybase.CreateAccount. As far as I can tell, this writes the new account info to the keybase store which is then read from each time maketx is used.

I was linking to a specific part of `gnokey add`, please ignore the other orchestration insanity that uses the keybase beforehand. I am trying to give you a snippet of how you can derive account keys from a raw mnemonic (and account path, in the form of account index and account number).

> How else would the data get into the keybase store with out using CreateAccount?

You are completely right, I totally forgot we initialize a keybase that is local to the txtar context 🤦‍♂️ 
https://github.com/gnolang/gno/blob/29fd2eadf2043d46934a3334b43e24eb33d1ec9f/gno.land/pkg/integration/testing_integration.go#L81

Thank you for pointing it out, there is no reason to derive accounts using a mnemonic like this when you already have a keybase instance for the txtar test 🙏